From 919ff298f2c9c43f4b809b5e51f169154b8a00e8 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 25 Mar 2024 14:30:49 +0200
Subject: [PATCH] [FSW-4100] fs1: pldm: init pll dividers only

On pldm emulator only the PLL dividers are implemented, the analog part
of the PLL is not emulated and the PLL is fed with the nominal clock.
Hence need to init the analog part of the PLL and wait for lock, only
dividers should be set.

Change-Id: Ia51ca2d27d7c4bc8c1bdb7b25daffebbdcd9fe0e
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/pldm.c b/arch/fs1/arc/pldm.c
index 5efc705..ed42b49 100644
--- a/arch/fs1/arc/pldm.c
+++ b/arch/fs1/arc/pldm.c
@@ -71,9 +71,17 @@
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
 
+#define PSOC_HBW_DIV_ID		0
+#define PSOC_LBW_DIV_ID		1
+#define PSOC_CFG_DIV_ID		2
+#define PSOC_UART_DIV_ID	3
+
 void hl_dev_psoc_pll_init(void)
 {
-	hl_pll_init(fs1_psoc_pll_cfg, 1);
+	hl_pll_reconfig_div(fs1_psoc_pll_cfg, PSOC_UART_DIV_ID);
+	hl_pll_reconfig_div(fs1_psoc_pll_cfg, PSOC_HBW_DIV_ID);
+	hl_pll_reconfig_div(fs1_psoc_pll_cfg, PSOC_LBW_DIV_ID);
+	hl_pll_reconfig_div(fs1_psoc_pll_cfg, PSOC_CFG_DIV_ID);
 }
 
 static bool is_cxl_mode(void)
